kong/tracing/propagation.lua

```diff
@@ -465,7 +464,7 @@ local function set(conf_header_type, found_header_type, proxy_span, conf_default
     set_header("uber-trace-id", fmt("%s:%s:%s:%s",
         to_hex(proxy_span.trace_id),
         to_hex(proxy_span.span_id),
-        to_hex(proxy_span.parent_id),
+        proxy_span.parent_id and to_hex(proxy_span.parent_id) or "0",
```

Comment:
```suggestion
        to_hex(proxy_span.parent_id or 0),
```

The same reason: to catch the incorrect type.

Reply:
It is by design, `to_hex("0")` will result in `30` which is not expected.

Reply:
It sounds like a problem that should be fixed like this:
```lua
to_hex(tonumber(proxy_span.parent_id or 0)),
```